gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -11,186 +11,232 @@ You may obtain a copy of the License at
+                args: ["{dataStore}", "{codeGenerator}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // userId, clientId, selectedPreferences
+            },
+            userHasAuthorized: {
+                funcName: "gpii.oauth2.authorizationService.userHasAuthorized",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // userId, clientId, redirectUri
+            },
+            exchangeCodeForAccessToken: {
+                funcName: "gpii.oauth2.authorizationService.exchangeCodeForAccessToken",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // code, clientId, redirectUri
+            },
+            getSelectedPreferences: {
+                funcName: "gpii.oauth2.authorizationService.getSelectedPreferences",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1"]
+                    // userId, authDecisionId
+            },
+            setSelectedPreferences: {
+                funcName: "gpii.oauth2.authorizationService.setSelectedPreferences",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // userId, authDecisionId, selectedPreferences
+            },
+            getUnauthorizedClientsForUser: {
+                funcName: "gpii.oauth2.authorizationService.getUnauthorizedClientsForUser",
+                args: ["{dataStore}", "{arguments}.0"]
+                    // userId
+            },
+            getAuthorizedClientsForUser: {
+                func: "{dataStore}.findAuthorizedClientsByUserId"
+                    // userId
+            },
+            revokeAuthorization: {
+                func: "{dataStore}.revokeAuthDecision"
+                    // userId, authDecisionId
+            },
+            getAccessTokenForOAuth2ClientIdAndGpiiToken: {
+                func: "{dataStore}.findAccessTokenByOAuth2ClientIdAndGpiiToken"
+                    // oauth2ClientId, gpiiToken
+            },
+            getAuthForAccessToken: {
+                func: "{dataStore}.findAuthByAccessToken"
+                    // accessToken
+            },
+            getClientByClientCredentialsAccessToken: {
+                func: "{dataStore}.findClientByClientCredentialsAccessToken"
+                    // accessToken
+            },
+            getClientCredentialsTokenPrivs: {
+                func: "{dataStore}.findClientCredentialsTokenPrivs"
+                    // accessToken
+            },
+            grantClientCredentialsAccessToken: {
+                funcName: "gpii.oauth2.authorizationService.grantClientCredentialsAccessToken",
+                args: ["{dataStore}", "{codeGenerator}", "{arguments}.0", "{arguments}.1"]
+                    // clientId, scope
+            },
+            savePrefs: { // TODO: This should be removed - see GPII-1521
```

Comment:
Will address this in a separate pull request using the JIRA ticket https://issues.gpii.net/browse/GPII-1521